users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -177,29 +243,71 @@ def get_masked():
         )
         actual_freq_mask_max_num = tf.minimum(freq_mask_max_num, total_freq_masks_max_size // freq_mask_max_size)
 
-        x_masked = random_mask(
-            x_masked,
-            batch_axis=data.batch_dim_axis,
-            axis=data.time_dim_axis,
-            min_num=0,
-            max_num=actual_time_mask_max_num,
-            max_dims=time_mask_max_size,
-            sorted_indices=tf.range(tf.shape(x)[data.time_dim_axis]),
-        )
-        x_masked = random_mask(
-            x_masked,
-            batch_axis=data.batch_dim_axis,
-            axis=data.feature_dim_axis,
-            min_num=0,
-            max_num=actual_freq_mask_max_num,
-            max_dims=freq_mask_max_size,
-            sorted_indices=sorted_indices,
-        )
+        if config["filter_based_masking_strategy"] == "variance":
+            f_resp = get_frequency_response(filter_layer)
+            variance = tf.math.reduce_variance(f_resp, axis=0)
+            n_features = tf.shape(x)[data.feature_dim_axis]
+            probs = tf.nn.softmax(variance)
+            uniform_probs = tf.ones_like(probs) / tf.cast(n_features, tf.float32)
+            final_probs = variance_factor * probs + (1 - variance_factor) * uniform_probs
+            enable_logging = tf.convert_to_tensor(config["enable_logging"], dtype=tf.bool)
+
+            def logging_ops():
+                with tf.control_dependencies(
+                    [
+                        tf.print(
+                            "Specaug variance Log: ",
+                            current_epoch,
+                            actual_time_mask_max_num,
+                            tf.shape(x)[data.time_dim_axis],
+                            variance.shape,
+                            sep=", ",
+                        )
+                    ]
+                ):
+                    return tf.identity(x_masked)
+
+            x_masked = tf.cond(enable_logging, logging_ops, lambda: tf.identity(x_masked))
+
+            x_masked = random_mask(
+                x_masked,
+                batch_axis=data.batch_dim_axis,
+                axis=data.time_dim_axis,
+                min_num=0,
+                max_num=actual_time_mask_max_num,
+                max_dims=time_mask_max_size,
+                sorted_indices=tf.range(tf.shape(x)[data.time_dim_axis]),
+            )
+            x_masked = filter_based_masking(
+                x_masked,
+                batch_axis=data.batch_dim_axis,
+                axis=data.feature_dim_axis,
+                probability_distribution=final_probs,
+                max_number_masks_for_probability_specaug=tf.gather(
+                    config["specaug_params"]["max_number_masks_for_variance_specaug"], current_epoch
+                ),
+            )
+        else:
```

Comment:
It should not be `else`. Just leave the code as it is and only add the `if` for the filter-based masking (then this if does also not contain time masking of course). If the regular masking is not desired, it should be set to 0 via the config options.